### DISCUSSION TITLE
Range Calendar onValueChange not firing

### DISCUSSION THREAD

**Discussion by MattHeslington** [melt-ui/melt-ui#731] [Q&A]
Hello,

I can't get the 'onValueChange' function to run when the values of the range calendar change. Is the function called something else? I had a similar problem before with the Select builder, and the 'change' function was 'onSelectedChange', not 'onSelectChange'. I've tried:

onValueChange
onValuesChange
onSelectedChange

```javascript
const updateFunction = ({ curr, prev }) => {
    console.log('rangeCalendar changed')
    return curr
}

const {
    elements: { calendar, heading, grid, cell, prevButton, nextButton },
    states: { months, headingValue, daysOfWeek, value },
    helpers: { isDateDisabled, isDateUnavailable }
} = createRangeCalendar({
    onValueChange: updateFunction
})
```
Any help very much appreciated.
Cheers, Matt

**Comment by garikAsplund** [comment]
That is strange. I can reproduce it for only range calendar by itself. The date range picker imports and uses the range calendar with a date range field and oddly enough the onValueChange works with that. I added an onValueChange argument to the dateRangeField props and that works, too. Can you use the date range picker builder instead? I think it has to do with the value store not being initiated in the props but not sure.

**Comment by TGlide** [comment]
An issue would be better suited here. Are you still facing problems?
